plugins/services/src/js/containers/service-connection/SDKServiceConnectionEndpointList.js

```diff
@@ -0,0 +1,265 @@
+      </ConfigurationMapSection>
+    );
+  }
+
+  getFileEndpoint(endpoint) {
+    return (
+      <ConfigurationMapRow key={endpoint.getName()}>
+        <ConfigurationMapLabel>
+          {endpoint.getName()}
+        </ConfigurationMapLabel>
+        <ConfigurationMapValue>
+          <a
+            className="active endpoint-download"
+            download={endpoint.getName()}
+            href={RouterUtil.getResourceDownloadPath(
+              "text/plain",
+              endpoint.getName(),
+              endpoint.getData()
+            )}
+          >
+            <span>
+              <Icon
+                id="download"
+                className="endpoint-download-icon"
+                size="mini"
+              />
+              {" "}
+              Download
+            </span>
+          </a>
+        </ConfigurationMapValue>
+      </ConfigurationMapRow>
+    );
+  }
+
+  getJSONEndpoints(endpoints) {
+    const jsonEndpoints = endpoints.filter(endpoint => {
+      return endpoint.isJSON();
+    });
+
+    if (jsonEndpoints.length === 0) {
+      return null;
+    }
+
+    return jsonEndpoints.map(endpoint => {
+      return this.getJSONEndpoint(endpoint);
+    });
+  }
+
+  getFileEndpointRows(fileEndpoints) {
+    return fileEndpoints.map(endpoint => {
+      return this.getFileEndpoint(endpoint);
+    });
+  }
+
+  getFileEndpoints(endpoints) {
+    const fileEndpoints = endpoints.filter(endpoint => {
+      return !endpoint.isJSON();
+    });
+
+    if (fileEndpoints.length === 0) {
+      return null;
+    }
+
+    return (
+      <ConfigurationMapSection>
+        <ConfigurationMapHeading>
+          Files
+        </ConfigurationMapHeading>
+        {this.getFileEndpointRows(fileEndpoints)}
+      </ConfigurationMapSection>
+    );
+  }
+
+  getAlertPanelSDKDeploying() {
+    return (
+      <div className="endpoint-sdk-deploying">
+        Endpoints for
+        {" "}
+        {this.props.service.getId()}
+        {" "}
+        will appear here once it is fully deployed and running.
+      </div>
+    );
+  }
+
+  render() {
+    const { actionDisabledModalOpen } = this.state;
+    const { service } = this.props;
+
+    if (actionDisabledModalOpen) {
+      return (
+        <ServiceActionDisabledModal
+          actionID={EDIT}
+          open={actionDisabledModalOpen}
+          onClose={this.handleOpenEditConfigurationModal.bind(this, false)}
+          service={service}
+        />
+      );
+    }
+
+    const sdkServiceEndpoints = SDKEndpointStore.getServiceEndpoints(
+      service.getId()
+    );
+
+    const sdkServiceError = SDKEndpointStore.getServiceError(service.getId());
+
+    const endpointsLoaded =
+      sdkServiceEndpoints &&
+      sdkServiceEndpoints.every(endpoint => {
+        return endpoint.endpointData;
+      });
+
+    if (!endpointsLoaded && !sdkServiceError) {
+      return <Loader />;
+    }
+
+    if (sdkServiceError) {
+      return this.getAlertPanelSDKDeploying();
```

Comment:
It feels wrong to render the alert panel SDK *deploying* in case of an error. We should be able to use the service state to either render `getAlertPanelSDKDeploying` or a proper error message.

Reply:
This is because when the server is deploying we receive an error state because it is not ready yet. The error message wouldn't be valuable for the user I think at this point.